File: Shantanu_Mamgain/Sliding-Window-Leetcode/Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int arr[] = new int[128];
        int start = 0;
        int end = 0;
        int maxLen = 0;
        while(end < s.length()) {      
            
            if(arr[s.charAt(end)] != start){
                start = Math.max(arr[s.charAt(end)], start);
            }
            arr[s.charAt(end)] = end +1;
            maxLen = Math.max(maxLen, end - start + 1);
            
            end ++;
             
            
        }

        return maxLen;
    }
}

File: Lakshay Garg/Leetcode/Sliding Window/Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size(),start=0,end=0,ans=0;
        if(n==0)return  0;
        if(n==1)return  1;
        vector<int>v(256,0);
        while(end<n){
            if(v[s[end]]>=start){
                start=v[s[end]];
            }
            v[s[end]]=end+1;
            ans=max(ans,end-start+1);
            end++;
        }
        return  ans;
    }
};
